world/shared_task_manager.cpp

```diff
@@ -182,7 +185,7 @@ void SharedTaskManager::AttemptSharedTaskCreation(
 	new_shared_task.SetMembers(request_members);
 
 	// add to shared tasks list
-	m_shared_tasks.emplace_back(new_shared_task);
+	auto& inserted = m_shared_tasks.emplace_back(new_shared_task);
```

Comment:
Curious why you use inserted this way instead of just passing `new_shared_task` to `HandleCompletedActivities` ?

Reply:
LockTask modifies the shared task with the `GetDbSharedTask` and `SetDbSharedTask` calls. I would have had to call it before it was inserted into m_shared_tasks to do that otherwise it would modify the local copy. (I originally had it like this but was playing with triggering completion after for tasks with no elements, but ended up removing that and leaving it after the emplace)

Reply:
Ahh interesting